### PR TITLE
Implement ATen Variable split and chunk.

### PR DESCRIPTION
**Pull request by gchanan**

### CODE REVIEW

**Comment by gchanan** [comment]
@colesbury as before, this includes the relevant ATen changes for testing and clarity purposes; I'll merge those through ATen before this is merged.

File: tools/autograd/templates/python_variable_methods.cpp [hunk 1]
```diff
@@ -35,6 +35,14 @@ inline PyObject* wrap(Scalar scalar) {
   return wrap(scalar.toTensor());
 }
 
+inline PyObject* wrap(TensorList tl) {
+  PyObject *tuple = PyTuple_New(tl.size());
```
Comment (colesbury) [inline tools/autograd/templates/python_variable_methods.cpp]: Use a `THPObjectPtr` to avoid leaks in-case one of the wraps fails below

**Review by colesbury — COMMENTED** [review]
lgtm

**Comment by gchanan** [comment]
this has been implemented in https://github.com/pytorch/pytorch/pull/3270.

**Pull request closed without merging**